File: cp_cpp/src/bin/buildinghighways.cpp
// Created by Ayush Biswas at 2026/05/05 20:32
// https://open.kattis.com/problems/buildinghighways
/// @head begin
#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>
/// @head end

/// @code begin
using namespace std;

void solve() {
    int n; cin >> n;
    vector<long long> v(n);
    for (int i = 0; i < n; i++) cin >> v[i];
    const auto m = *ranges::min_element(v);
    const auto s = accumulate(v.begin(), v.end(), 0LL);
    cout << m * (n - 2) + s;
}

int main() {
    cin.tie(nullptr);

    solve();
}
/// @code end

File: cp_rust/src/bin/buildinghighways.rs
// Created by Ayush Biswas at 2026/05/06 19:37
// https://open.kattis.com/problems/buildinghighways
use cp_lib::*;

// @code begin
use cpio::*;

sol! {
    fn solution(
        n: usize,
        a: [usize]
    ) -> usize {
        let b = a.iter().min().unwrap();
        a.iter().sum::<usize>() + (n - 2) * b
    }
}
// @code end
